src/WebdriverClassicDriver.php

```diff
@@ -1,21 +1,1390 @@
+        } catch (Throwable $e) {
+            throw new DriverException("Could not start driver: {$e->getMessage()}", 0, $e);
+        }
+    }
+
+    /**
+     * Sets the timeouts to apply to the webdriver session
+     *
+     * @param array $timeouts The session timeout settings: Array of {script, implicit, page} => time in milliseconds
+     * @throws DriverException
+     * @api
+     */
+    public function setTimeouts(array $timeouts): void
+    {
+        $this->timeouts = $timeouts;
+
+        if ($this->isStarted()) {
+            $this->applyTimeouts();
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
     public function isStarted(): bool
     {
-        return false;
+        return $this->webDriver !== null;
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws DriverException
+     */
+    public function stop(): void
+    {
+        if (!$this->webDriver) {
+            throw new DriverException('Driver has not been started');
+        }
+
+        try {
+            $this->getWebDriver()->quit();
+            $this->webDriver = null;
+        } catch (Throwable $e) {
+            throw new DriverException('Could not close connection', 0, $e);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws DriverException
+     */
+    public function reset(): void
+    {
+        $this->getWebDriver()->manage()->deleteAllCookies();
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws DriverException
+     */
+    public function visit($url): void
+    {
+        $this->getWebDriver()->navigate()->to($url);
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws DriverException
+     */
+    public function getCurrentUrl(): string
+    {
+        return $this->getWebDriver()->getCurrentURL();
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws DriverException
+     */
+    public function reload(): void
+    {
+        $this->getWebDriver()->navigate()->refresh();
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws DriverException
+     */
+    public function forward(): void
+    {
+        $this->getWebDriver()->navigate()->forward();
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws DriverException
+     */
+    public function back(): void
+    {
+        $this->getWebDriver()->navigate()->back();
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws DriverException
+     */
+    public function switchToWindow($name = null): void
+    {
+        if ($name === null) {
+            $name = $this->initialWindowName;
+        }
+
+        if (is_string($name)) {
+            $name = $this->getWindowHandleFromName($name);
+        }
+
+        $this->getWebDriver()->switchTo()->window($name);
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws DriverException
+     */
+    public function switchToIFrame($name = null): void
+    {
+        $frameQuery = $name;
+        if ($name && is_string($name) && $this->getWebDriver()->isW3cCompliant()) {
```

Comment:
this `is_string($name)` is useless. As the argument is `string|null`, it will always be true here.